yesod-core/src/Yesod/Core/Types.hs

```diff
@@ -306,6 +306,20 @@ data Content = ContentBuilder !BB.Builder !(Maybe Int) -- ^ The content and opti
              | ContentFile !FilePath !(Maybe FilePart)
              | ContentDontEvaluate !Content
 
+contentToTruncatedString :: Content -> Int -> String
+contentToTruncatedString (ContentBuilder builder maybeLength) maxLength =
+    let
+      truncated = take maxLength (show builder)
+      excess = case maybeLength of
+        (Just length) -> length - maxLength
+        Nothing -> 0
+    in case (excess > 0) of
+      True -> truncated ++ "... (+" ++ show excess ++ ")"
+      False -> truncated
+contentToTruncatedString (ContentSource _) _ = "ContentSource"
+contentToTruncatedString (ContentFile _ _) _ = "ContentFile"
+contentToTruncatedString (ContentDontEvaluate _) _ = "ContentDontEvaluate"
```

Comment:
This function is exported. Can we add a doc string with @since annotation?

```suggestion
-- | doc me up boss
--
-- @since 1.6.28.0
contentToTruncatedString :: Content -> Int -> String
contentToTruncatedString (ContentBuilder builder maybeLength) maxLength =
    let
      truncated = take maxLength (show builder)
      excess = case maybeLength of
        (Just length) -> length - maxLength
        Nothing -> 0
    in case (excess > 0) of
      True -> truncated ++ "... (+" ++ show excess ++ ")"
      False -> truncated
contentToTruncatedString (ContentSource _) _ = "ContentSource"
contentToTruncatedString (ContentFile _ _) _ = "ContentFile"
contentToTruncatedString (ContentDontEvaluate _) _ = "ContentDontEvaluate"
```